cmd/agent/app/integrations.go

```diff
@@ -40,6 +42,54 @@ var (
 	tufConfig    string
 )
 
+type integrationVersion struct {
+	major int
+	minor int
+	fix   int
+}
+
+func parseVersion(version string) (*integrationVersion, error) {
+	var major, minor, fix int
+	if version == "" {
+		return nil, nil
+	}
+	_, err := fmt.Sscanf(version, "%d.%d.%d", &major, &minor, &fix)
+	if err != nil {
+		return nil, fmt.Errorf("unable to parse version string %s: %v", version, err)
+	}
+	return &integrationVersion{major, minor, fix}, nil
+}
+
+func (v *integrationVersion) String() string {
+	return fmt.Sprintf("%d.%d.%d", v.major, v.minor, v.fix)
+}
+
+func (v *integrationVersion) isAbove(otherVersion *integrationVersion) bool {
+	if otherVersion == nil {
+		return true
+	}
+
+	if v.major > otherVersion.major {
+		return true
+	} else if v.major < otherVersion.major {
+		return false
+	}
+
+	if v.minor > otherVersion.minor {
+		return true
+	} else if v.minor < otherVersion.minor {
+		return false
+	}
+
+	if v.fix > otherVersion.fix {
+		return true
+	} else if v.fix < otherVersion.fix {
+		return false
+	}
+
+	return true
```

Comment:
This will cause a suboptimal installation to happen later, as I will explain below. I think this function should not return true on equality, otherwise it violates the `>` expectation. Right now it's `>=`.